source/SemVerAnalyzer/Engine/SemverExtensions.cs

```diff
@@ -8,66 +8,25 @@ public static class SemverExtensions
 	{
 		static readonly Regex _versionFormat = new Regex(@"^(?<major>[0-9]+)\.(?<minor>[0-9]+)\.(?<patch>[0-9]+)((?<prerelease>-\w*)|(?<trailer>\.0))?(?<githash>\+[0-9a-f]{40})?$");
 
-		public static int MajorVersion(this string version)
-		{
-			var match = _versionFormat.Match(version);
-			if (!match.Success) throw new FormatException("Not a semantic version");
-
-			var major = int.Parse(match.Groups["major"].Value);
-
-			return major;
-		}
-
-		public static int MinorVersion(this string version)
-		{
-			var match = _versionFormat.Match(version);
-			if (!match.Success) throw new FormatException("Not a version");
+		public static int MajorVersion(this string version) => version.ToSemver().Major;
+		public static int MinorVersion(this string version) => version.ToSemver().Minor;
+		public static int PatchVersion(this string version) => version.ToSemver().Patch;
 
-			var minor = int.Parse(match.Groups["minor"].Value);
+		public static string GetSemVer(string version) => version.ToSemver().ToString();
 
-			return minor;
-		}
+		public static string GetSuggestedVersion(this string version, VersionBumpType bump) => version.ToSemver().GetSuggestedVersionAfterBump(bump);
 
-		public static int PatchVersion(this string version)
-		{
+		public static Semver ToSemver(this string version){
 			var match = _versionFormat.Match(version);
 			if (!match.Success) throw new FormatException("Not a version");
-
-			var patch = int.Parse(match.Groups["patch"].Value);
-
-			return patch;
-		}
-
-		public static string GetSemVer(string version)
-		{
-			var match = _versionFormat.Match(version);
-			if (!match.Success) throw new FormatException("Not a version");
-
+			
 			var major = int.Parse(match.Groups["major"].Value);
 			var minor = int.Parse(match.Groups["minor"].Value);
 			var patch = int.Parse(match.Groups["patch"].Value);
-
-			return $"{major}.{minor}.{patch}";
+			var prerelease = match.Groups["prerelease"]?.Value;
+			var trailer = match.Groups["trailer"]?.Value;
```

Comment:
`group.Value` will always be set, and if `group.Success` is `false` then it's set to `""` (which removes the chance for nulls in ToString anyway).

Reply:
...so you don't need the `?`..